editor/plugins/tiles/tile_map_layer_editor.cpp

```diff
@@ -1558,40 +1532,22 @@ int TileMapLayerEditorTilesPlugin::_get_transformed_alternative(int p_alternativ
 	bool transform_transpose = p_alternative_id & TileSetAtlasSource::TRANSFORM_TRANSPOSE;
 
 	switch (p_transform) {
-		case TRANSFORM_ROTATE_LEFT:
-		case TRANSFORM_ROTATE_RIGHT: {
-			// A matrix with every possible flip/transpose combination, sorted by what comes next when you rotate.
-			const LocalVector<bool> rotation_matrix = {
-				// NOLINTBEGIN(modernize-use-bool-literals)
-				0, 0, 0,
-				0, 1, 1,
-				1, 1, 0,
-				1, 0, 1,
-				1, 0, 0,
-				0, 0, 1,
-				0, 1, 0,
-				1, 1, 1
-				// NOLINTEND(modernize-use-bool-literals)
-			};
-
-			for (int i = 0; i < 8; i++) {
-				if (transform_flip_h == rotation_matrix[i * 3] && transform_flip_v == rotation_matrix[i * 3 + 1] && transform_transpose == rotation_matrix[i * 3 + 2]) {
-					if (p_transform == TRANSFORM_ROTATE_LEFT) {
-						i = i / 4 * 4 + (i + 1) % 4;
-					} else {
-						i = i / 4 * 4 + Math::posmod(i - 1, 4);
-					}
-					transform_flip_h = rotation_matrix[i * 3];
-					transform_flip_v = rotation_matrix[i * 3 + 1];
-					transform_transpose = rotation_matrix[i * 3 + 2];
-					break;
-				}
-			}
+		case TRANSFORM_ROTATE_LEFT: { // (h, v, t) -> (v, !h, !t)
+			bool negated_flip_h = !transform_flip_h;
+			transform_flip_h = transform_flip_v;
+			transform_flip_v = negated_flip_h;
+			transform_transpose = !transform_transpose;
+		} break;
+		case TRANSFORM_ROTATE_RIGHT: { // (h, v, t) -> (!v, h, !t)
+			bool negated_flip_v = !transform_flip_v;
+			transform_flip_v = transform_flip_h;
+			transform_flip_h = negated_flip_v;
+			transform_transpose = !transform_transpose;
```

Comment:
That just a simplification, should behave the same.

Reply:
Tested it. Seems to work fine indeed.